### PR TITLE
Fix regression #479-bis

### PR DESCRIPTION
**Pull request by jimklimov**
Restores the failsafe removed quickly in #480, this time applying proper variable naming to the checked item.

### CODE REVIEW

File: lib/ZnapZend/Config.pm [hunk 1]
```diff
@@ -80,15 +80,30 @@ my $checkBackupSets = sub {
...
         }
 
+        if ( $backupSet->{src} =~ m/[\@]/ ) {
+            # If we are here, somebody fed us a snapshot in the list of
+            # datasets, which is likely a bug elsewhere in discovery.
+            # We do not want to fail whole backup below due to faulted
+            # dataSetExists() below, so just ignore this entry.
+            # If we really do get here, take a hard look at recursive
+            # and/or inherited modes for run-once.
+            print STDERR "#checkBackupSets# SKIP backupSet='" . $backupSet->{src} .
```
Comment (oetiker) [inline lib/ZnapZend/Config.pm:101]: not using zLog here ?

Comment (jimklimov) [reply lib/ZnapZend/Config.pm:101]: I thought about it a little... this PR is essentially a `git revert` + typo fix, so I did not go too deep. Given that this catches a situation that should never happen at all, I am not sure what could be proper logging here.

Comment (oetiker) [reply lib/ZnapZend/Config.pm:101]: well the kind of logging that is best suited to catch the attendents attention ... zLog->error I would suggest ... after all they should be watching the znapzend logs

Comment (jimklimov) [reply lib/ZnapZend/Config.pm]: Done, checked that `$self->zLog->error()` by itself is not fatal to code progressing ;)

**Comment by jimklimov** [comment]
Note: the check for snapshot in this block may be among the targets for #514 to check for bookmarks as another "unexpected" case... But that is a separate issue to investigate and tackle globally.

**Pull request merged**